### ISSUE TITLE
Type error: Argument 3 passed to Codeception\Event\FailEvent::__construct() must be an instance of Exception, none given,

### ISSUE THREAD

**Issue by scoops** [Codeception/Codeception#3999]
I've been working on an application in Yii 2 and trying to write tests as I go. I hadn't made any changes to my application since last Wednesday, but this morning I got an error when running my tests. I tried updating Codeception (the copy local to the project, and the global copy on my machine) and still received the same error (though the formatting was different). 

Here's the output I got:

```

  [Symfony\Component\Debug\Exception\FatalThrowableError]
  Type error: Argument 3 passed to Codeception\Event\FailEvent::__construct() must be an instance of Exception, none given, called in C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\codeception\codeception\src\Codeception\
  PHPUnit\Listener.php on line 50


Exception trace:
 () at C:\WAMP\htdocs\ktos_upgrade\vendor\codeception\codeception\src\Codeception\Event\FailEvent.php:16
 Codeception\Event\FailEvent->__construct() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\codeception\codeception\src\Codeception\PHPUnit\Listener.php:50
 Codeception\PHPUnit\Listener->addError() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\phpunit\phpunit\src\Framework\TestResult.php:250
 PHPUnit_Framework_TestResult->addError() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\phpunit\phpunit\src\Framework\TestResult.php:821
 PHPUnit_Framework_TestResult->run() at C:\WAMP\htdocs\ktos_upgrade\vendor\phpunit\phpunit\src\Framework\TestCase.php:926
 PHPUnit_Framework_TestCase->run() at C:\WAMP\htdocs\ktos_upgrade\vendor\phpunit\phpunit\src\Framework\TestSuite.php:728
 PHPUnit_Framework_TestSuite->run() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\codeception\codeception\src\Codeception\PHPUnit\Runner.php:97
 Codeception\PHPUnit\Runner->doEnhancedRun() at C:\WAMP\htdocs\ktos_upgrade\vendor\codeception\codeception\src\Codeception\SuiteManager.php:153
 Codeception\SuiteManager->run() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\codeception\codeception\src\Codeception\Codecept.php:207
 Codeception\Codecept->runSuite() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\codeception\codeception\src\Codeception\Codecept.php:176
 Codeception\Codecept->run() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\codeception\codeception\src\Codeception\Command\Run.php:316
 Codeception\Command\Run->runSuites() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\codeception\codeception\src\Codeception\Command\Run.php:285
 Codeception\Command\Run->runIncludedSuites() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\codeception\codeception\src\Codeception\Command\Run.php:248
 Codeception\Command\Run->execute() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\symfony\console\Command\Command.php:261
 Symfony\Component\Console\Command\Command->run() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\symfony\console\Application.php:817
 Symfony\Component\Console\Application->doRunCommand() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\symfony\console\Application.php:185
 Symfony\Component\Console\Application->doRun() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\symfony\console\Application.php:116
 Symfony\Component\Console\Application->run() at C:\Users\rmli223.AD\AppData\Roaming\Composer\vendor\codeception\codeception\codecept:28

```

* Codeception version: 2.1.11
* PHP Version: 7.0.11
* Operating System: Windows 10
* Installation type: Composer
* List of installed packages (`composer show`):
```
behat/gherkin                       v4.4.5   Gherkin DSL parser for PHP 5.3
bower-asset/bootstrap               v3.3.7   The most popular front-end framework for developing responsive, mobile first projects on the web.
bower-asset/jquery                  2.2.4
bower-asset/jquery.inputmask        3.3.4    jquery.inputmask is a jquery plugin which create an input mask.
bower-asset/punycode                v1.3.2
bower-asset/typeahead.js            v0.11.1
bower-asset/yii2-pjax               v2.0.6
cebe/markdown                       1.1.1    A super fast, highly extensible markdown parser for PHP
codeception/codeception             2.2.9    BDD-style testing framework
codeception/specify                 0.4.6    BDD code blocks for PHPUnit and Codeception
codeception/verify                  0.3.2    BDD assertion library for PHPUnit
doctrine/instantiator               1.0.5    A small, lightweight utility to instantiate objects in PHP without invoking their constructors
ezyang/htmlpurifier                 v4.8.0   Standards compliant HTML filter written in PHP
facebook/webdriver                  1.3.0    A PHP client for WebDriver
fzaninotto/faker                    v1.6.0   Faker is a PHP library that generates fake data for you.
guzzlehttp/guzzle                   6.2.2    Guzzle is a PHP HTTP client library
guzzlehttp/promises                 v1.3.1   Guzzle promises library
guzzlehttp/psr7                     1.3.1    PSR-7 message implementation
kartik-v/yii2-krajee-base           v1.8.7   Base library and foundation components for all Yii2 Krajee extensions.
kartik-v/yii2-widget-datetimepicker v1.4.3   Enhanced Yii2 wrapper for the bootstrap datetimepicker plugin (sub repo split from yii2-widgets)
myclabs/deep-copy                   1.6.0    Create deep copies (clones) of your objects
phpdocumentor/reflection-common     1.0      Common reflection classes used by phpdocumentor to reflect the code structure
phpdocumentor/reflection-docblock   3.1.1    With this component, a library can provide support for annotations via DocBlocks or otherwise retrieve information that is embedded in a DocBlock.
phpdocumentor/type-resolver         0.2.1
phpspec/php-diff                    v1.1.0   A comprehensive library for generating differences between two hashable objects (strings or arrays).
phpspec/prophecy                    v1.6.2   Highly opinionated mocking framework for PHP 5.3+
phpunit/php-code-coverage           4.0.5    Library that provides collection, processing, and rendering functionality for PHP code coverage information.
phpunit/php-file-iterator           1.4.2    FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-text-template           1.2.1    Simple template engine.
phpunit/php-timer                   1.0.8    Utility class for timing
phpunit/php-token-stream            1.4.9    Wrapper around PHP's tokenizer extension.
phpunit/phpunit                     5.7.11   The PHP Unit Testing framework.
phpunit/phpunit-mock-objects        3.4.3    Mock Object library for PHPUnit
psr/http-message                    1.0.1    Common interface for HTTP messages
psr/log                             1.0.2    Common interface for logging libraries
sebastian/code-unit-reverse-lookup  1.0.0    Looks up which function or method a line of code belongs to
sebastian/comparator                1.2.4    Provides the functionality to compare PHP values for equality
sebastian/diff                      1.4.1    Diff implementation
sebastian/environment               2.0.0    Provides functionality to handle HHVM/PHP environments
sebastian/exporter                  2.0.0    Provides the functionality to export PHP variables for visualization
sebastian/global-state              1.1.1    Snapshotting of global state
sebastian/object-enumerator         2.0.0    Traverses array structures and object graphs to enumerate all referenced objects
sebastian/recursion-context         2.0.0    Provides functionality to recursively process PHP variables
sebastian/resource-operations       1.0.0    Provides a list of PHP built-in functions that operate on resources
sebastian/version                   2.0.1    Library that helps with managing the version number of Git-hosted PHP projects
swiftmailer/swiftmailer             v5.4.5   Swiftmailer, free feature-rich PHP mailer
symfony/browser-kit                 v3.2.3   Symfony BrowserKit Component
symfony/console                     v3.2.3   Symfony Console Component
symfony/css-selector                v3.2.3   Symfony CssSelector Component
symfony/debug                       v3.2.3   Symfony Debug Component
symfony/dom-crawler                 v3.2.3   Symfony DomCrawler Component
symfony/event-dispatcher            v3.2.3   Symfony EventDispatcher Component
symfony/finder                      v3.2.3   Symfony Finder Component
symfony/polyfill-mbstring           v1.3.0   Symfony polyfill for the Mbstring extension
symfony/process                     v3.2.3   Symfony Process Component
symfony/yaml                        v3.2.3   Symfony Yaml Component
webmozart/assert                    1.2.0    Assertions to validate method input/output with nice error messages.
yiisoft/yii2                        2.0.11.1 Yii PHP Framework Version 2
yiisoft/yii2-bootstrap              2.0.6    The Twitter Bootstrap extension for the Yii framework
yiisoft/yii2-composer               2.0.5    The composer plugin for Yii extension installer
yiisoft/yii2-debug                  2.0.7    The debugger extension for the Yii framework
yiisoft/yii2-faker                  2.0.3    Fixture generator. The Faker integration for the Yii framework.
yiisoft/yii2-gii                    2.0.5    The Gii extension for the Yii framework
yiisoft/yii2-swiftmailer            2.0.6    The SwiftMailer integration for the Yii framework

```
* Suite configuration:

```yml
include:
  - codeception/common
  - codeception/console
  - codeception/backend
  - codeception/frontend

paths:
  log: codeception/_output

settings:
  colors: true

modules:
  config:
      Yii2:
          configFile: 'codeception/config/config-local.php'
```

**Comment by Naktibalda** [comment]
You have 2 different minor versions of Codeception installed
> Codeception version: 2.1.11

>codeception/codeception             2.2.9    BDD-style testing framework

Please don't mix global and local installations.
Prefer a local one.

It should be fixed by #3997

**Comment by scoops** [comment]
That fixed the issue. Thank you!
